packages/app/src/cli/services/deploy.ts

```diff
@@ -57,45 +56,21 @@ export async function deploy(options: DeployOptions) {
   outputInfo(`Deploying your work to Shopify Partners. It will be part of ${partnersApp.title}`)
   outputNewline()
 
-  const extensions = await Promise.all(
-    options.app.extensions.ui.map(async (extension) => {
-      return {
-        uuid: identifiers.extensions[extension.localIdentifier]!,
-        config: JSON.stringify(await extension.deployConfig()),
-        context: '',
-      }
-    }),
-  )
-  if (useThemebundling()) {
-    const themeExtensions = await Promise.all(
-      options.app.extensions.theme.map(async (extension) => {
-        return {
-          uuid: identifiers.extensions[extension.localIdentifier]!,
-          config: '{"theme_extension": {"files": {}}}',
-          context: '',
-        }
-      }),
-    )
-    extensions.push(...themeExtensions)
-  }
```

Comment:
deploy.ts no longer builds the config for each extension, that's a responsibility of the ExtensionInstance depending on the features.